RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegmentBuilder.cc

```diff
@@ -0,0 +1,381 @@
+/** 
+ * \file GEMCSCSegmentBuilder.cc
+ *
+ */
+
+#include <RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegmentBuilder.h>
+#include <RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegmentAlgorithm.h>
+#include <RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegmentBuilderPluginFactory.h>
+
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include <FWCore/Utilities/interface/Exception.h>
+#include <FWCore/MessageLogger/interface/MessageLogger.h> 
+
+#include <Geometry/Records/interface/MuonGeometryRecord.h>
+#include <Geometry/CSCGeometry/interface/CSCChamberSpecs.h>
+#include <Geometry/CSCGeometry/interface/CSCLayer.h>
+#include <Geometry/CSCGeometry/interface/CSCGeometry.h>
+#include <Geometry/GEMGeometry/interface/GEMGeometry.h>
+#include <Geometry/GEMGeometry/interface/GEMEtaPartition.h>
+
+#include <DataFormats/MuonDetId/interface/CSCDetId.h>
+#include <DataFormats/MuonDetId/interface/GEMDetId.h>
+#include <DataFormats/CSCRecHit/interface/CSCRecHit2D.h>
+#include <DataFormats/CSCRecHit/interface/CSCRangeMapAccessor.h>
+#include <DataFormats/CSCRecHit/interface/CSCRecHit2DCollection.h>
+#include <DataFormats/CSCRecHit/interface/CSCSegmentCollection.h>
+#include <DataFormats/CSCRecHit/interface/CSCSegment.h>
+#include <DataFormats/GEMRecHit/interface/GEMRecHit.h>
+#include <DataFormats/GEMRecHit/interface/GEMRecHitCollection.h>
+
+
+GEMCSCSegmentBuilder::GEMCSCSegmentBuilder(const edm::ParameterSet& ps) : gemgeom_(0), cscgeom_(0) 
+{
+
+    // Algo name
+    std::string algoName = ps.getParameter<std::string>("algo_name");
+    edm::LogVerbatim("GEMCSCSegmentBuilder")<< "[GEMCSCSegmentBuilder :: ctor] algorithm name: " << algoName;
+
+    // SegAlgo parameter set 	  
+    edm::ParameterSet segAlgoPSet = ps.getParameter<edm::ParameterSet>("algo_psets");
+ 
+    // Ask factory to build this algorithm, giving it appropriate ParameterSet
+    algo = GEMCSCSegmentBuilderPluginFactory::get()->create(algoName, segAlgoPSet);
+
+}
+
+GEMCSCSegmentBuilder::~GEMCSCSegmentBuilder() 
+{
+    delete algo;
+    edm::LogVerbatim("GEMCSCSegmentBuilder")<<"[GEMCSCSegmentBuilder :: dstor] deleted the algorithm";
+}
+
+void GEMCSCSegmentBuilder::LinkGEMRollsToCSCChamberIndex(const GEMGeometry* gemGeo, const CSCGeometry* cscGeo) 
+{
+
+  for (TrackingGeometry::DetContainer::const_iterator it=gemGeo->dets().begin();it<gemGeo->dets().end();it++)
+    {
+      if(dynamic_cast< const GEMChamber* >( *it ) != 0 )
+	{
+	  const GEMChamber* ch = dynamic_cast< const GEMChamber* >( *it );
```

Comment:
Lines 59 and 61 are in reverse order. Define the ch variable, and then check it with if (ch != 0).